### PR TITLE
Prep for v1.11.1

### PR DESCRIPTION
**Pull request by odow**
## Pre-release

 - [x] Check that the pinned packages in `docs/Project.toml` are updated. We pin
       the versions so that changes in the solvers (changes in printing, small
       numeric changes) do not break the printing of the JuMP docs in arbitrary
       commits.
 - [x] Check that the `rev` fields in `docs/packages.toml` are updated. We pin
       the versions of solvers and extensions to ensure that changes to their
       READMEs do not break the JuMP docs in arbitrary commits, and to ensure
       that the versions are compatible with the latest JuMP and
       MathOptInterface releases.
 - [x] Update `docs/src/changelog.md`
 - [x] Run https://github.com/jump-dev/JuMP.jl/actions/workflows/extension-tests.yml
        using a `workflow_dispatch` trigger to check for any changes in JuMP that
        broke extensions.
          * https://github.com/jump-dev/JuMP.jl/actions/runs/4942011171
 - [x] Change the version number in `Project.toml`
 - [x] Update the links in README.md
 - [x] The commit messages in this PR do not contain `[ci skip]`
       (See [Documenter#965](https://github.com/JuliaDocs/Documenter.jl/issues/965).)

## The release

 - [ ] After merging this pull request, comment `@JuliaRegistrator register` in
       the GitHub commit. This should automatically publish a new version to the
       Julia registry, as well as create a tag, and rebuild the documentation
       for this tag.

       These steps can take quite a bit of time (1 hour or more), so don't be
       surprised if the new documentation takes a while to appear. In addition,
       the links in the README will be broken until JuliaHub fetches the new
       version on their servers.

## Post-release

 - [ ] Once the tag is created, update the relevant `release-` branch. The latest
       release branch at the time of writing is `release-1.0` (we haven't
       back-ported any patches that needed to create a `release-1.Y` branch). To
       to update the release branch with the v1.10.0 tag, do:
       ```
       git checkout release-1.0
       git pull
       git merge v1.10.0
       git push
       ```

### CODE REVIEW

**Comment by odow** [comment]
cc @pulsipher do you have time to look at InfiniteOpt with the latest `master`?

I'm seeing lots of errors: https://github.com/jump-dev/JuMP.jl/actions/runs/4942011171/jobs/8835171512

I don't know what would have caused the issue on our end.

**Comment by pulsipher** [comment]
Hmmm, I'll take a closer look tonight or early tomorrow (EST).

**Comment by odow** [comment]
It seems this is actually a Julia v1.9 issue. I'll keep digging

**Comment by odow** [comment]
InfiniteOpt confirmed as unrelated to JuMP. Some weird bug on Julia v1.9...

File: docs/packages.toml [hunk 1]
```diff
@@ -146,7 +146,7 @@
   rev = "44e0bff587208fb4e0779db440f4104bc0bba2ec"
 [SDDP]
     user = "odow"
-    rev = "v1.1.4"
+    rev = "v1.4.0"
```
Comment (odow) [inline docs/packages.toml:149]:
```suggestion
    rev = "v1.5.0"
```

File: docs/src/changelog.md [hunk 2]
```diff
@@ -7,6 +7,24 @@ CurrentModule = JuMP
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Version 1.11.1 (May 12, 2023)
```
Comment (odow) [inline docs/src/changelog.md:10]:
```suggestion
## Version 1.11.1 (May 15, 2023)
```

**Comment by odow** [comment]
@blegat happy?

File: docs/src/changelog.md [hunk 3]
```diff
@@ -7,6 +7,24 @@ CurrentModule = JuMP
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Version 1.11.1 (May 15, 2023)
```
Comment (odow) [inline docs/src/changelog.md:10]:
```suggestion
## Version 1.11.1 (May 17, 2023)
```

**Review by blegat — APPROVED** [review]

**Pull request merged**